src/command/process_update_task.rs

```diff
@@ -162,46 +293,239 @@ impl UpdateProcessor {
+                    return self.not_valid_offset_message().await;
+                }
+
+                self.repo
+                    .modify_offset(&self.chat.id, self.chat.user_id, offset)
+                    .await?;
+
+                // we have city_id , hour and minutes well formatted stored in selected see in process_time func.
+                let vec: Vec<&str> = self.chat.selected.as_ref().unwrap().split(',').collect();
+                // This unwraps are safe because we know that is stored in the correct format.
+                let city_id = vec[0].parse::<i32>().unwrap();
+
+                let time: Vec<&str> = vec[1].split(':').collect();
+                let user_hour = time[0].parse::<i8>().unwrap();
+                let minutes = time[1].parse::<i8>().unwrap();
+
+                self.schedule_forecast(offset, city_id, user_hour, minutes)
+                    .await?;
+                Ok(())
+            }
+
+            Err(_) => self.not_valid_offset_message().await,
+        }
+    }
+
+    async fn not_time_message(&self) -> Result<(), BotError> {
+        self.cancel(Some(
+            "That's not a well formatted time, it has to be formatted with this format `hour:minutes` being hour a number in range [0,23] 
+            and minutes a number in range [0,59]. The command was cancelled"
+            .to_string(),
+        ))
+        .await
+    }
+
+    fn parse_time(hour_or_minutes: &str, max_range: i8, min_range: i8) -> Result<i8, ()> {
+        match hour_or_minutes.parse::<i8>() {
+            Ok(number) => {
+                if !(min_range..=max_range).contains(&number) {
+                    Err(())
+                } else {
+                    Ok(number)
+                }
+            }
+            Err(_) => Err(()),
+        }
+    }
+
+    async fn process_time(&self) -> Result<(), BotError> {
+        let vec: Vec<&str> = self.text.trim().split(':').collect();
 
-            ClientState::SetCity => self.set_default_city(city).await,
+        if vec.len() != 2 {
+            return self.not_time_message().await;
+        }
 
-            _ => self.revert_state().await,
+        let hour = match Self::parse_time(vec[0], 23, 0) {
+            Err(_) => return self.not_time_message().await,
+            Ok(number) => number,
+        };
+
+        let minutes = match Self::parse_time(vec[1], 59, 0) {
+            Err(_) => return self.not_time_message().await,
+            Ok(number) => number,
+        };
+
+        // In selected we have city_id captured in scheduled_city_number func
+
+        match self.chat.offset {
+            Some(offset) => {
+                self.schedule_forecast(
+                    offset,
+                    self.chat.selected.as_ref().unwrap().parse::<i32>().unwrap(),
+                    hour,
+                    minutes,
+                )
+                .await?;
+            }
+            None => {
+                self.repo
```

Comment:
I think it's error prone, 

because you're assuming a user will run the offset command after the time command.

what if he will have something else instead of the selected time in the selected field

Reply:
I am not sure if its an error. I have tested this, and seems to work fine.

Please discuss in #36